pkg/porter/schema.go

```diff
@@ -0,0 +1,111 @@
+package porter
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/deislabs/porter/pkg/config"
+
+	"github.com/pkg/errors"
+)
+
+func (p *Porter) PrintManifestSchema() error {
+	schemaMap, err := p.GetManifestSchema()
+	if err != nil {
+		return err
+	}
+
+	schema, err := json.MarshalIndent(&schemaMap, "", "  ")
+	if err != nil {
+		return errors.Wrap(err, "could not marshal the composite porter manifest schema")
+	}
+
+	fmt.Fprintln(p.Out, string(schema))
+	return nil
+}
+
+func (p *Porter) GetManifestSchema() (map[string]interface{}, error) {
+	b, err := p.schemas.Find("manifest.json")
+	if err != nil {
+		return nil, err
+	}
+
+	manifestSchema := make(map[string]interface{})
+	err = json.Unmarshal(b, &manifestSchema)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not unmarshal the root porter manifest schema")
+	}
+
+	definitionSchema, ok := manifestSchema["definitions"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("root porter manifest schema has invalid definitions type, expected map[string]interface{} but got %T", manifestSchema["definitions"])
+	}
+
+	propertiesSchema, ok := manifestSchema["properties"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("root porter manifest schema has invalid properties type, expected map[string]interface{} but got %T", manifestSchema["properties"])
+	}
+
+	mixinSchema, ok := propertiesSchema["mixins"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("root porter manifest schema has invalid properties.mixins type, expected map[string]interface{} but got %T", propertiesSchema["mixins"])
+	}
+
+	itemsSchema, ok := mixinSchema["items"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("root porter manifest schema has invalid properties.mixins.items type, expected map[string]interface{} but got %T", mixinSchema["items"])
+	}
+
+	enumSchema, ok := itemsSchema["enum"].([]interface{})
+	if !ok {
+		return nil, errors.Errorf("root porter manifest schema has invalid properties.mixins.items.enum type, expected []interface{} but got %T", itemsSchema["enum"])
+	}
+
+	installSchema, ok := propertiesSchema["install"].(map[string]interface{})
```

Comment:
Do we also need `upgradeSchema` and `uninstallSchema`?

I ran `porter schema` off of this branch and see: https://gist.github.com/vdice/114bec04b6bfbf7dabf14027372e591a

Or, to put it another way, I currently don't see the other actions of `upgrade` and `uninstall` represented in the generated schema for mixins that currently implement both (`exec` and `helm`).

Reply:
The plan was to let Ivan get his hands on this, try it out for install, and if it works, then finish the other two in a follow-up.